src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -194,52 +194,95 @@ public void importEntries(List<BibEntry> entries) {
     }
 
     public void importEntryWithDuplicateCheck(BibDatabaseContext bibDatabaseContext, BibEntry entry) {
-        ImportCleanup cleanup = new ImportCleanup(bibDatabaseContext.getMode());
-        BibEntry cleanedEntry = cleanup.doPostCleanup(entry);
-        BibEntry entryToInsert = cleanedEntry;
+        BibEntry cleanedEntry = cleanUpEntry(bibDatabaseContext, entry);
+        Optional<BibEntry> existingDuplicateInLibrary = findDuplicate(bibDatabaseContext, cleanedEntry);
 
-        Optional<BibEntry> existingDuplicateInLibrary = new DuplicateCheck(Globals.entryTypesManager).containsDuplicate(bibDatabaseContext.getDatabase(), entryToInsert, bibDatabaseContext.getMode());
+        Optional<BibEntry> entryToInsert = Optional.of(cleanedEntry);
```

Comment:
What happend here? Scrolling down, the Optinal is not needed. Chains of `.get()` are an indication that something is wrong... (`.ifpresent(...)` is sometimes right. Here, normal objects are better. Please find attached the suggestins. You can use the GitHub UI to commit the suggestions)

```suggestion
        BibEntry entryToInsert = cleanedEntry;
```